### DISCUSSION TITLE
Stuck on Please Wait. Loading...

### DISCUSSION THREAD

**Discussion by adamesq** [rrirower/DP2SceneGraphWizard#37] [Q&A]
I've been using an MRSS feed from JWPlayer since maybe 2017 for my channel. (The JSON feed for JWPlayer has never worked.) I tried doing Roku's conversion. I side loaded the channel with the MRSS feed. All went well (except categories didn't show...just one long feed of content.) I got all the way through packaging...then ran into an issue with Deep Linking. Entering one or 2 items into the Deep Linking section on the dashboard got rid of the error, but I'm concerned that hitting Publish will cause the channel to go POOF.

SOOOO, I'm trying DP2SceneGraphWizard. This results in the channel starting to boot (via side loading), then getting Please Wait. Loading...

I'm thinking I have a feed issue (since it's MRSS). I've tried plugging one of my videos in through Roku's JSON template. The feed validates on jsonschemavalidator.net and Roku reported back that the feed validates. HOWEVER, redirecting the DP2SceneGraphWizard's manifest to the validated feed still gives me the splash screen followed by "Please Wait. Loading..." and going through Roku's own conversion with this feed results in the splash screen, then Channel background with no content.

Can someone please give me some suggestions here? I'm just trying to see where the issue is and get a working feed at this point.

**Comment by rrirower** [comment]
Please provide a URL for your feed and I'll take a look. My guess is that there's a problem with the feed.

**Reply by adamesq** [reply]
Thank you very much for the quick reply, @rrirower. 
The feed I've been using for years (still works for DP) and used with DP2SceneGraphWizard is https://content.jwplatform.com/feeds/bvnULZlM.rss

In trying to rebuild, I've been testing with https://silverwerks.tv/r/FromScratchTest.json (trying to learn from Roku's examples).

I'm not sure if, even if this test JSON is correct, that I can just plug it in to the manifest generated from the MRSS feed. I'm also noticing that both the simple and advanced samples from Roku don't have an example URL for a video file.🤔 I feel like I'm stumbling around here in the dark a bit!

**Reply by rrirower** [reply]
@adamesq I took a look at your files.  I ran the RSS file in debug.  Here's what I found.

- Your JSON file is not a feed file.  It's actually a Search feed.

- Your RSS file does not conform to the [Roku MRSS feed spec](https://developer.roku.com/docs/specs/direct-publisher-feed-specs/mrss-dp-spec.md) in that the code used by the wizard tool uses 'media:category' tags to bucket your content.  Your file does not contain those tags.

The best option for you is to obtain a JSON feed file if possible.  If that's not possible, make sure your feed file contains "media:category" tags.

**Reply by adamesq** [reply]
Thanx again @rrirower. I'm working through it and don't mind building a JSON from Roku's sample. However, I tried plugging the unaltered, working Roku SceneGraph Master Sample into the manifest that I had generated DP2SceneGraphWizard. The site goes blank after the splash screen going that route. 

I'm looking at a few paths:
1) Build backwards from the Roku's SceneGraph master sample since that side-loads properly
2) Try WP Smart TV Plugin (it would serve the purpose of helping populate a new WP install for the brand anyway). I'm just not sure how well this plugin works with the current specifications???
3) Poke around in JWP's JSON feed, which I forgot to send in my post above. This should be the JSON version of the MRSS feed: 
https://cdn.jwplayer.com/v2/playlists/bvnULZlM

THANK YOU again for taking the time to help & reply!

**Reply by rrirower** [reply]
The JSON file you posted in '3' above is not valid for the Roku JSON spec.  Keep in mind, that the DP2SceneGraph wizard tool requires the feed content to be compliant with the Roku specs for JSON or MRSS.  The tool expects certain requirements to be met and if they are not, your channel will not operate correctly.  Please see the [prerequisites](https://github.com/rrirower/DP2SceneGraphWizard) for the tool.

**Reply by adamesq** [reply]
I've tested the sample feed from Roku (mentioned in 1 above) - https://github.com/rokudev/scenegraph-master-sample. For some reason, when I use DP2SceneGraphWizard](https://github.com/rrirower/DP2SceneGraphWizard). I have it temporarily at https://silverwerks.tv/r/roku-developers-feed-vsw.json (only changed provider name). I get past the splash page to the background, but empty channel. 

Any ideas on why this isn't working?

**Reply by rrirower** [reply]
A quick glance appears to show that the feed file does not contain any Category objects. I'll look closer tomorrow.

**Reply by adamesq** [reply]
Very odd...it's the main feed for all of the Roku sample work. Thanx for looking...and have a good night!

**Reply by rrirower** [reply]
@adamesq The feed file you posted does not contain Category objects.  However, the feed may be appropriate for the channel code that loads it.  Keep in mind, that the wizard code was written to make the transition from DP to SceneGraph easy for those without technical backgrounds.  It leverages the DP JSON and MRSS Schema Specs so that the user does not have to change his feed file.  That means that the feed file MUST contain Category objects since they are used to define the rows in the Grid Screen UI.

**Reply by adamesq** [reply]
GOT IT! Thank you! That's helpful. At about 1:30am, I got into Feed Manager. THAT gave me a working feed! 👏🏻🙏🏻🎉 LOTS to fill in from my old MRSS...I think I had 99+ items in there. Since I'm on a Mac, I went on a diversion and figured out how to create an Amazon EC2 Windows instance to be able to use your tools! THANK YOU for being such a big help to the community! (Just updated to the new alpha 7 release. Advertising is absolutely something I'd like to explore. I'll try that after adding more to my feed.)

Off-hand, would you be able to point me in a direction to figure out how modify the channel to have a hero row? I'm a little hesitant to poke around the very old hero sample code in Roku's GitHub since their current course sample feed is doing something odd with categories that makes it not work with DP2SGW whereas Feed Manager finally got me to a working one. (Their feed didn't have Category objects, but it the course they use is pulling it from the feed without the category tag:
```
	"providerName": "Roku Developers",
	"language": "en-US",
	"lastUpdated": "2020-03-15T02:01:00+02:00",
	"shortFormVideos": [
. . . .then further down. . . 
	"series": [{
	
```
"shortformvideos" and "series" can be called whatever and show up as rows without the category tag...don't know if this is legit even though it seems to work when side loading the samples.)

**Reply by rrirower** [reply]
I've been asked for several other UI enhancements.  I don't have a Hero layout figured out yet since it's low on the priority list.

**Reply by adamesq** [reply]
Again...REALLY appreciate you sharing your work and insights!
